docs/graphs/classdiagram_lib.dot

```diff
@@ -0,0 +1,34 @@
+digraph hierarchy {
+size="5,5"
+node[   shape=record,
+        style=filled,
+        fillcolor=gray95]
+edge[dir=back, arrowtail=none]
+
+
+ps[label = "{ProgramState|
```

Comment:
I'm not yet sure what `ProgramState` is. 

Maybe it's the state of the entire program. In that case I have two comments:
1. If it should really describe the state of the whole program, we'd probably need more data fields than `all_transactions` and `filtered_transactions`.
1. The state of the program is probably comprised of more components. I guess we should focus on those lower level components first before designing such a high level data structure as the `ProgramState`.

Or maybe it's the state of all the transactions. In this case I'd prefer another name for it, e.g. `TransactionState`.

Reply:
You're right, the name ProgramState is bad for a couple of reasons
1. It's not the whole program, most of the program state is probably part of the frontend
2. Within the lib, there is not much "state" left, when you think of it. Actually if we just generate views on the data, then this would also be better handled in the frontend, and everything else is handled by the all-powerful dataframe, so for the lib, there is not much left except passing data around from frontend to DB. 
So, the program state is just where the data is stored, after it has been loaded from the database - and maybe is a facade for the lib - but that's debatable

Reply:
So a better name would maybe be `TransactionContainer`? Let's first be clear about what it does, then for the name